### ISSUE TITLE
DoTs being clipped

### ISSUE THREAD

**Issue by MullichWhitemane** [gamefreedomgit/Maelstrom#2202]
Original report: https://discord.com/channels/1040715551531618365/1077239759828697268

Relevant copy pastes: https://wowpedia.fandom.com/wiki/Damage_over_Time
"Also as of patch 4.0.1, DoTs no longer can be 'clipped'. Previously, if a DoT was refreshed before it had expired, the tick timer was reset, causing a delay of up to double the normal time between ticks between the last tick that occurred and the first tick of the refreshed DoT. Now, the tick timer is constant. If a DoT is refreshed before it expires, the duration of the DoT will be extended by up to the full normal duration of the DoT, and ticks will continue to occur at the prior interval."

The following screenshot was taken after refreshing a Corruption before it's very last tick - as you can see, Corruption failed to tick 8 times, only ticking the standard 7 times. I've also tested with with many other DoTs - Immolate, Bane of Agony, Shadow Word: Pain and Living Bomb to name a few examples.
![image](https://user-images.githubusercontent.com/121984461/220819017-e55f5f8f-2aca-4d9f-b5bd-20c0a7bafeae.png)

**Comment by UltraNix** [comment]
Fixed.

**Comment by UltraNix** [comment]
No as it's a not a DOT.

**Comment by MullichWhitemane** [comment]
I was made aware of new info in this new discord report: https://discord.com/channels/1040715551531618365/1078680179485986846

The following is a normal Corruption from a Warlock, with an 18 second duration and when the ticks occur.
![corruption](https://user-images.githubusercontent.com/121984461/221221071-4a2b842c-877c-4dac-a4c0-4611bf28b949.png)

The following is that same Corruption being refreshed on it's last tick, and that last tick is then moved to the new Corruption - this is what was fixed.
![corruption2](https://user-images.githubusercontent.com/121984461/221221289-cb0ae6a3-c61b-43f4-a599-e626c1d9d555.png)

And the following is what should happen no matter which tick you refresh corruption on - in the example image, it was refreshed before the 4th tick.
![corruption3](https://user-images.githubusercontent.com/121984461/221221424-1ef0879f-874c-42a2-9c54-a466f10eba7f.png)

Tl;dr: No matter which DoT Tick you refresh a DoT on, it should always take a tick out of the old DoT.
Also to note: Bane of Doom seems unable to be clipped, even on it's last tick.
